src/Components/Aspire.Azure.Storage.Queues/AzureStorageQueueSettings.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Data.Common;
+using Aspire.Azure.Common;
+
+namespace Aspire.Azure.Storage.Queues;
+
+/// <summary>
+/// Provides the client configuration settings for connecting to Azure Storage queue.
+/// </summary>
+public sealed class AzureStorageQueueSettings : AzureStorageQueuesSettings, IConnectionStringSettings
+{
+    /// <summary>
+    ///  Gets or sets the name of the blob container.
+    /// </summary>
+    public string? QueueName { get; set; }
+
+    void IConnectionStringSettings.ParseConnectionString(string? connectionString)
+    {
+        if (string.IsNullOrEmpty(connectionString))
+        {
+            return;
+        }
+
+        DbConnectionStringBuilder builder = new() { ConnectionString = connectionString };
+
+        // NOTE: if ever these contants are changed, the AzureQueueStorageResource in Aspire.Hosting.Azure.Storage class should be updated as well.
+        if (builder.TryGetValue("Endpoint", out var endpoint) && builder.TryGetValue("QueueName", out var queueName))
```

Comment:
Why do they both have to be set before either property is populated. I would expect these checks to be separated out. That way the Enpoint can come on the connection string, and the QueueName is configured in code during the `configureSettings` callback.

Also, we should consider the "base" `AzureStorageQueuesSettings` ParseConnectionString logic as well. If the connection string is just a Url, use it as the `ServiceUri`.

Reply:
Because we expect the connection string to be in the exact format we defined in `AzureQueueStorageResource`. If it's in a different format, then it's something else.

> If the connection string is just a Url, use it as the ServiceUri.

Can you show me the example?